frontend/src/components/problem/ProblemDisplay.tsx

```diff
@@ -142,141 +233,204 @@ function ProblemDisplay(props: ProblemDisplayParams) {
   };
 
   return (
-    <Content>
-      <MediumText>Name:</MediumText>
-      <TextInput
-        name="name"
-        value={newProblem.name}
-        onChange={handleChange}
-      />
-
-      <MediumText>Description:</MediumText>
-      <StyledMarkdownEditor
-        defaultValue={newProblem.description}
-        onChange={(getNewValue) => handleDescriptionChange(getNewValue())}
-      />
-
-      <MediumText>Difficulty:</MediumText>
-      {Object.keys(Difficulty).map((key) => {
-        const difficulty = Difficulty[key as keyof typeof Difficulty];
-        if (difficulty !== Difficulty.Random) {
-          return (
-            <DifficultyButton
-              onClick={() => handleEnumChange('difficulty', difficulty)}
-              active={difficulty === newProblem.difficulty}
-              enabled
-            >
-              {key}
-            </DifficultyButton>
-          );
-        }
-        return null;
-      })}
-
-      <MediumText>Problem Inputs:</MediumText>
-      {newProblem.problemInputs.map((input, index) => (
-        <div>
-          <Text bold>{`Input ${index + 1}`}</Text>
-          <TextInput
-            value={newProblem.problemInputs[index].name}
-            onChange={(e) => handleInputChange(index,
-              e.target.value, newProblem.problemInputs[index].type)}
+    <>
+      <MainContent>
+        <FlexBareContainer>
+          <SmallHeaderText>Problem</SmallHeaderText>
+          <InvertedSmallButtonLinkAutoLeftMargin
+            onClick={() => onClick(newProblem)}
+            to="/problems/all"
+          >
+            Back
+          </InvertedSmallButtonLinkAutoLeftMargin>
+          <SmallButton
+            onClick={() => onClick(newProblem)}
+          >
```

Comment:
Since a bit of extra work was necessary to get the auto left margin to work (namely the createButtonLink function), maybe an alternative simpler way (to cut down on extra code) would be if we put these two buttons into a div and make that aligned/flex/margin right? Then the back button could just become an InvertedSmallButtonLink using the original creatButtonLink function

I haven't tested this though and sometimes flex stuff can be a hassle, so nbd if we keep this as-is for now though

Reply:
That's definitely much smarter - thanks! Updated by creating a wrapper container with `margin-left: auto` 👍